batch-models/src/main/scala/org/ekstep/analytics/dashboard/CuratedCollectionsModel.scala

```diff
@@ -0,0 +1,138 @@
+package org.ekstep.analytics.dashboard
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.functions._
+import DashboardUtil._
+import org.apache.spark.sql.types.{ArrayType, StringType, StructType}
+import org.ekstep.analytics.framework.{AlgoInput, AlgoOutput, FrameworkContext, IBatchModelTemplate, Output}
+
+import java.io.Serializable
+
+case class CCDummyInput(timestamp: Long) extends AlgoInput  // no input, there are multiple sources to query
+case class CCDummyOutput() extends Output with AlgoOutput  // no output as we take care of kafka dispatches ourself
+
+case class CCConfig( debug: String,
+                     sparkElasticsearchConnectionHost: String, cassandraKeyspace: String, cassandraTable: String, sparkDruidRouterHost: String
+                   ) extends Serializable
+
+object CuratedCollectionsModel extends IBatchModelTemplate[String, CCDummyInput, CCDummyOutput, CCDummyOutput] with Serializable{
+
+  implicit val className: String = "org.ekstep.analytics.dashboard.CuratedCollectionsModel"
+
+  override def name() = "CuratedCollectionsModel"
+
+  override def preProcess(events: RDD[String], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[CCDummyInput] = {
+    // we want this call to happen only once, so that timestamp is consistent for all data points
+    val executionTime = System.currentTimeMillis()
+    sc.parallelize(Seq(CCDummyInput(executionTime)))
+  }
+
+  override def algorithm(data: RDD[CCDummyInput], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[CCDummyOutput] = {
+    val timestamp = data.first().timestamp // extract timestamp from input
+    implicit val spark: SparkSession = SparkSession.builder.config(sc.getConf).getOrCreate()
+    processCuratedCollectionsData(timestamp, config)
+    sc.parallelize(Seq()) // return empty rdd
+  }
+
+  override def postProcess(events: RDD[CCDummyOutput], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[CCDummyOutput] = {
+    sc.parallelize(Seq())  // return empty rdd
+  }
+
+  def processCuratedCollectionsData(timestamp: Long, config: Map[String, AnyRef])(implicit spark: SparkSession, sc: SparkContext, fc: FrameworkContext): Unit = {
+    // parse model config
+    println(config)
+    implicit val conf: CCConfig = parseConfig(config)
+    if (conf.debug == "true") debug = true // set debug to true if explicitly specified in the config
+
+    val curatedCollectionDataDf = curatedCollectionReportDf()
+    curatedCollectionDataDf.repartition(1).write.option("header", true).csv("tmp/output/CuratedCollectionReport")
```

Comment:
Can we make it such that this spark job can output to kafka as well as the disk depending on configuration?